pkg/kubectl/autoscale.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+const (
+	DefaultCPUUtil   = 80
+	scaleSubResource = "scale"
+)
+
+type HorizontalPodAutoscalerV1Beta1 struct{}
+
+func (HorizontalPodAutoscalerV1Beta1) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"default-name", true},
+		{"name", false},
+		{"scaleRef-kind", false},
+		{"scaleRef-namespace", false},
+		{"scaleRef-name", false},
+		{"scaleRef-apiVersion", false},
+		{"scaleRef-subresource", false},
+		{"min", true},
+		{"max", true},
+		{"cpu%", false},
```

Comment:
--cpu-percent, the flag, should not have a default value. It should let the server apply the default if not specified.

With respect to a specific flag (--cpu-percent) vs. generic flags (--percent-utilization=...):

The API now has a specific field for cpu, not just a generic list or map. There are pragmatic reasons why we did that, such as punting on the format of names for custom metrics and similarity to GCE autoscaling APIs. However, I believe there's a strong technical reason, also. 

CPU and memory are very different from custom metrics in that we can build a deeper understanding of them into the system. We're going to need to take multiple signals (not just "usage") into account in order to do a good job, especially when combined with vertical auto-scaling, QoS, oversubscription, usage-based scheduling, etc. Whether/how we'll need to change/extend the API is not certain at this point, but it's useful to convey that they're special, IMO.

--request and --limit are different. They already create a 2x2 matrix. 4 flags would be tedious, and we know that we'll need to add more resources soon, such as disk space, iops, network ingress/egress bandwidth, gpus, etc. While, I expect these resources to have uniform behavior for request and limit, I don't expect to autoscale using many/any of these resources. Moving to application-level metrics seems like a better way to go.